decentraland/screens/EthereumSignUpScreen.js

```diff
@@ -9,19 +9,15 @@ import {
   showError,
   fundAccountWithEthers,
   fundAccountWithMana,
+  createAccount,
 } from "./helpers";
 
-import { Account } from "tasit-sdk";
-
 export class EthereumSignUpScreen extends React.Component {
   _onboarding = async () => {
     try {
       const { setAccount, setSetupInProgress } = this.props;
 
-      // Note: The timeout for account creation is about ~10 secs.
-      // See more: https://github.com/tasitlabs/tasit/issues/42
-      const account = Account.create();
-      setAccount(account);
+      const account = await createAccount();
```

Comment:
Are we making sure that they only get to this step on the `EthereumSignUpScreen` if they don't already have an account?

Reply:
Yes, the Sign-up/Sign-in flow are called only if the account isn't setted:

**from BuyLandScreen:**
```javascript
  _onBuy = landForSale => {
    try {
      const { accountInfo } = this.props;
      const { account } = accountInfo;
      if (!account) this._setupAccount();
      // The _buy function assumes that the account is created, funded and allowed
      // The user will get back to this else case after going through an account setup flow
      else this._buy(landForSale);
    } catch (err) {
      showError(err);
    }
  };
```

Reply:
Ah right - okay, perfect. Thanks for explaining that! Resolving this.